lib/puppet.rb

```diff
@@ -80,7 +79,7 @@ class << self
         else
           GettextSetup.initialize(locale_path, :file_format => :mo)
         end
-        FastGettext.locale = GettextSetup.negotiate_locale(Locale.current.language)
+        GettextSetup.negotiate_locale!(GettextSetup.candidate_locales)
```

Comment:
The method with the `!` just calls the non-mutating one and assigns the result to FastGettext the way we were doing it previously. The main difference for us is using `candidate_locales` instead of `Locale.current`.